server/lib/availabilitySync.ts

```diff
@@ -234,62 +183,115 @@ class AvailabilitySync {
     } while (mediaPage.length > 0);
   }
 
+  private async findMediaStatus(
+    requests: MediaRequest[],
+    is4k?: boolean
+  ): Promise<MediaStatus> {
+    let filteredRequests: MediaRequest[] = requests;
+
+    if (is4k !== undefined) {
+      filteredRequests = requests.filter((request) => request.is4k === is4k);
+    }
+
+    let mediaStatus: MediaStatus;
+
+    if (
+      filteredRequests.some(
+        (request) => request.status === MediaRequestStatus.APPROVED
+      )
+    ) {
+      mediaStatus = MediaStatus.PROCESSING;
+    } else if (
+      filteredRequests.some(
+        (request) => request.status === MediaRequestStatus.PENDING
+      )
+    ) {
+      mediaStatus = MediaStatus.PENDING;
+    } else {
+      mediaStatus = MediaStatus.UNKNOWN;
+    }
+
+    return mediaStatus;
+  }
+
```

Comment:
New reusable function created which will sort through the requests to see if any are still approved or pending. This is used on the top level of the mediaExists and mediaUpdater function.